trieste/utils/__init__.py

```diff
@@ -13,4 +13,4 @@
 # limitations under the License.
 """ This package contains library utilities. """
 from . import objectives
-from .misc import jit, shapes_equal, to_numpy
+from .misc import Err, Ok, Result, T_co, jit, shapes_equal, to_numpy
```

Comment:
Is `T_co` actually needed outside of the `misc` package? seems a slightly weird export to me.

Reply:
ah, yes, it's needed for the docs. We don't put every module in the docs (to keep both import paths and the docs simple). The `misc` module is one of those, so we need to import everything that needs to appear in the API